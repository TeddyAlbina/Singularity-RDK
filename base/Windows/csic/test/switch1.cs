class T {
	public static void f(int x) {
		switch (x) {
			case 1: x = 1; goto case 4;
			case -2: x = 2; break;
			default: System.Console.WriteLine(x); break;
			case 4: x = 5; goto default;
		}
	}
	public static void Main() {
		f(1);
	}
}