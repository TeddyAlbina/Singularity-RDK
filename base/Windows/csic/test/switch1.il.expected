.assembly 'switch1' { }
.class 'T'
	extends [mscorlib]System.Object
{
.method hidebysig public static void 'f'(int32 'x') {
ldarg 0 // x
.locals init ([0]int32 't0')
stloc 0 // t0
ldloc 0 // t0
ldc.i4 1
sub
switch (
$4,
$1,
$1,
$6,
)
ldloc 0 // t0
ldc.i4 -2
beq $5
br $1
$4:
ldc.i4 1
starg 0 // x
br $6
$5:
ldc.i4 2
starg 0 // x
br $3
$1:
ldarg 0 // x
call  void [mscorlib]System.Console::'WriteLine'(int32)
br $3
$6:
ldc.i4 5
starg 0 // x
br $1
$2:
$3:
ret
.maxstack 2
} // end of method T.f
.method hidebysig public static void 'Main'() {
.entrypoint
ldc.i4 1
call  void 'T'::'f'(int32)
ret
.maxstack 1
} // end of method T.Main
.method hidebysig specialname rtspecialname public void .ctor() {
ldarg.0
call instance void [mscorlib]System.Object::.ctor()
ret
.maxstack 1
} // end of constructor T.T
} // end of class T
